src/sentry/models/outbox.py

```diff
@@ -94,22 +89,274 @@ class OutboxCategory(IntEnum):
-    ORGANIZATION_MEMBER_CREATE = 13  # Unused
+    UNUSUED_THREE = 13
     SEND_SIGNAL = 14
     ORGANIZATION_MAPPING_CUSTOMER_ID_UPDATE = 15
     ORGAUTHTOKEN_UPDATE = 16
     PROVISION_ORGANIZATION = 17
     POST_ORGANIZATION_PROVISION = 18
-    SEND_MODEL_SIGNAL = 19
+    UNUSED_ONE = 19
     DISABLE_AUTH_PROVIDER = 20
     RESET_IDP_FLAGS = 21
     MARK_INVALID_SSO = 22
     SUBSCRIPTION_UPDATE = 23
 
+    AUTH_PROVIDER_UPDATE = 24
+    AUTH_IDENTITY_UPDATE = 25
+
     @classmethod
     def as_choices(cls):
         return [(i.value, i.value) for i in cls]
 
+    def connect_region_model_updates(self, model: Type[ProcessUpdatesWithRegionOutboxes]) -> None:
+        def receiver(
+            object_identifier: int,
+            payload: Optional[Mapping[str, Any]],
+            shard_identifier: int,
+            *args,
+            **kwds,
+        ):
+            from sentry.receivers.outbox import maybe_process_tombstone
+
+            maybe_instance: ProcessUpdatesWithRegionOutboxes | None = maybe_process_tombstone(
+                cast(Any, model), object_identifier, region_name=None
+            )
+            if maybe_instance is None:
+                model.handle_async_deletion(
+                    identifier=object_identifier, shard_identifier=shard_identifier, payload=payload
+                )
+            else:
+                maybe_instance.handle_async_replication(shard_identifier=shard_identifier)
+
+        process_region_outbox.connect(receiver, weak=False, sender=self)
+
+    def connect_control_model_updates(self, model: Type[ProcessUpdatesWithControlOutboxes]) -> None:
+        def receiver(
+            object_identifier: int,
+            payload: Optional[Mapping[str, Any]],
+            shard_identifier: int,
+            region_name: str,
+            *args,
+            **kwds,
+        ):
+            from sentry.receivers.outbox import maybe_process_tombstone
+
+            maybe_instance: ProcessUpdatesWithControlOutboxes | None = maybe_process_tombstone(
+                cast(Any, model), object_identifier, region_name=region_name
+            )
+            if maybe_instance is None:
+                model.handle_async_deletion(
+                    identifier=object_identifier,
+                    region_name=region_name,
+                    shard_identifier=shard_identifier,
+                    payload=payload,
+                )
+            else:
+                maybe_instance.handle_async_replication(
+                    shard_identifier=shard_identifier, region_name=region_name
+                )
+
+        process_control_outbox.connect(receiver, weak=False, sender=self)
+
+    def get_scope(self) -> OutboxScope:
+        for scope_int, categories in _outbox_categories_for_scope.items():
+            if self not in categories:
+                continue
+            break
+        else:
+            raise KeyError
+        return OutboxScope(scope_int)
+
+    def as_region_outbox(
```

Comment:
Convenience methods!  Generating an outbox becomes a lot simpler since the shard can be generally inferred given the OutboxCategory and model instance.